src/borg/archiver.py

```diff
@@ -1121,11 +1121,22 @@ def item_to_tarinfo(item, original_path):
     def do_diff(self, args, repository, manifest, key, archive):
         """Diff contents of two archives"""
 
+        def actual_change(j):
+            if j["type"] != "modified":
+                return True
+            else:
+                # Added/removed keys will not exist if chunker params differ
+                # between the two archives. Err on the side of caution and assume
+                # a real modification in this case (short-circuiting retrieving
+                # non-existent keys).
+                return not {"added", "removed"} <= j.keys() or not (j["added"] == 0 and j["removed"] == 0)
+
         def print_json_output(diff, path):
-            print(json.dumps({"path": path, "changes": [j for j, str in diff]}, sort_keys=True, cls=BorgJsonEncoder))
+            print(json.dumps({"path": path, "changes": [j for j, str in diff if actual_change(j)]}, sort_keys=True, cls=BorgJsonEncoder))
 
         def print_text_output(diff, path):
-            print("{:<19} {}".format(' '.join([str for j, str in diff]), path))
+            print(diff)
```

Comment:
Was that for debugging while developing this?

Reply:
Indeed... oops :D.